src/metabase/driver/mongo/query_processor.clj

```diff
@@ -211,16 +234,45 @@
 
 ;;; ### filter
 
+;; omf: 'max' date in a date range is next day at 00:00:00.000
+(defn- resolve-between [basetype minvalue maxvalue]
+  (cond (= basetype :type/DateTime)
+    {$gte minvalue
+     $lt (t/plus maxvalue (t/days 1))}
+  :else
+    {$gte minvalue
+     $lte maxvalue})
+)
+
+;; omf: a single day ends before next day at 00:00:00.000
+(defn- resolve-equal [basetype value unit]
+  (cond (= basetype :type/DateTime)
+    {$gte value
+     $lt (case unit
+           :year (t/plus value (t/years 1))
+           :month (t/plus value (t/months 1))
+           :week (t/plus value (t/weeks 1))
+           (t/plus value (t/days 1)))}
+  :else
+    {"$eq" value})
+)
+
 (defn- parse-filter-subclause [{:keys [filter-type field value] :as filter} & [negate?]]
-  (let [field (when field (->lvalue field))
+  ;; omf: Maybe there is a better way to do this. 'field' is being received as field: {...} or as field:{field: {...}}
+  ;; We need basetype so we can do a specific setup for 'between' and '=' for datetimes
+  ;; We need unit so we can do a specific setup for '=' for relative datetimes
+  (let [basetype (when field (or (:base-type field) (:base-type (:field field))))
```

Comment:
Something like `(:base-type nil)` just returns `nil` so no need to wrap everything in `when` here.